src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -327,15 +334,47 @@ internal PSVersionHashTable(IEqualityComparer equalityComparer) : base(equalityC
 
         /// <summary>
         /// Returns ordered collection with Keys of 'PSVersionHashTable'
+        /// We want see special order:
+        ///     1. PSVersionName
+        ///     2. PSEditionName
+        ///     3. Remaining properties in alphabetical order
         /// </summary>
         public override ICollection Keys
         {
             get
             {
-                Array arr = new string[base.Keys.Count];
-                base.Keys.CopyTo(arr, 0);
-                Array.Sort(arr, StringComparer.OrdinalIgnoreCase);
-                return arr;
+                ArrayList keyList = new ArrayList(base.Keys);
+                keyList.Sort(new PSVersionTableComparer());
+                return keyList;
+            }
+        }
+
+        private class PSVersionTableComparer : Comparer<object>
+        {
+            public override int Compare(object x, object y)
+            {
+                string xString = (string)LanguagePrimitives.ConvertTo(x, typeof(string), null);
+                string yString = (string)LanguagePrimitives.ConvertTo(y, typeof(string), null);
+                if (PSVersionInfo.PSVersionName == xString)
+                {
+                    return -1;
+                }
+                else if (PSVersionInfo.PSVersionName == yString)
+                {
+                    return 1;
+                }
+                else if (PSVersionInfo.PSEditionName == xString)
+                {
+                    return -1;
+                }
+                else if (PSVersionInfo.PSEditionName == yString)
+                {
+                    return 1;
+                }
+                else
+                {
+                    return xString.CompareTo(yString);
```

Comment:
Maybe use `String.Compare(string strA, string strB, System.StringComparison comparisonType)` so that you can specify `StringComparison.OrdinalIgnoreCase`.